File: rrouter-core/src/main/java/org/sheedon/rrouter/DataCloneable.java
package org.sheedon.rrouter;

/**
 * 数据拷贝
 *
 * @Author: sheedon
 * @Email: sheedonsun@163.com
 * @Date: 2021/11/2 6:02 下午
 */
public interface DataCloneable extends Cloneable{

    Object clone() throws CloneNotSupportedException;
}

File: rrouter-core/src/main/java/org/sheedon/rrouter/core/DataCloneable.kt
/*
 * Copyright (C) 2022 Sheedon.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package org.sheedon.rrouter.core

import kotlin.Throws

/**
 * 数据拷贝
 *
 * @Author: sheedon
 * @Email: sheedonsun@163.com
 * @Date: 2021/11/2 6:02 下午
 */
interface DataCloneable : Cloneable {
    @Throws(CloneNotSupportedException::class)
    public override fun clone(): Any
}